test/erlcloud_cloudformation_tests.erl

```diff
@@ -0,0 +1,544 @@
+        fun describe_stack_events_input_tests/1
+    ]}.
+
+start() ->
+    meck:new(erlcloud_aws),
+    ok.
+
+stop(_) ->
+    meck:unload(erlcloud_aws).
+
+%%==============================================================================
+%% Output Test helpers
+%%==============================================================================
+
+output_test(Fun, {Line, {Description, Response, Result}}) ->
+    {Description, {Line, fun() ->
+                        meck:expect(erlcloud_aws, aws_request_xml4, 6,
+                               {ok, element(1, xmerl_scan:string(binary_to_list(Response)))}
+                            ),
+                        Actual = Fun(),
+                        ?assertEqual(Result, Actual)
+                    end}}.
+
+
+%%==============================================================================
+%% Input Test helpers
+%%==============================================================================
+
+validate_param(_Param = {Key, _Value}, Params) ->
+    true = proplists:is_defined(Key, Params).
+
+validate_params(Params, Expected) ->
+    [validate_param(X, Params)
+        || X <- [{"Action", something}, {"Version", something} | Expected]
+
+    ].
+input_expect(Response, Params) ->
+
+    fun(post, "cloudformation.us-east-1.amazonaws.com", "/", QParams, "cloudformation", _) ->
+        validate_params(QParams, Params),
+        Response
+    end.
+
+input_test(Response, {Line, {Description, Fun, Params}}) when is_list(Description) ->
+
+    InputFunction = input_expect(Response, Params),
+
+    meck:expect(erlcloud_aws, aws_request_xml4, InputFunction),
+
+    {Description, {Line, fun() ->
+            Fun()
+        end}}.
+
+
+%%==============================================================================
+%% Input Tests
+%%==============================================================================
+
+list_stacks_all_input_tests(_) ->
+    Response = {ok, element(1, xmerl_scan:string(
+            binary_to_list(<<"<ListStacksResponse>null</ListStacksResponse>">>)
+        ))},
+
+    ExpectedParams = [],
+
+    input_test(Response, {?LINE, {"Test list all stacks input",
+        fun() ->
+            erlcloud_cloudformation:list_stacks_all([], #aws_config{})
+        end, ExpectedParams}}).
+
+list_stack_resources_input_tests(_) ->
+    Response = {ok, element(1, xmerl_scan:string(
+        binary_to_list(<<"<ListStackResourcesResponse>null</ListStackResourcesResponse>">>)
+    ))},
+
+    ExpectedParams = [{"StackName", something}],
+
+    input_test(Response, {?LINE, {"Test list all stack resources",
+        fun() ->
+            erlcloud_cloudformation:list_stack_resources_all(ExpectedParams, #aws_config{})
+        end, ExpectedParams}}).
+
+get_template_summary_input_tests(_) ->
+    Response = {ok, element(1, xmerl_scan:string(
+        binary_to_list(<<"<GetTemplateSummaryResponse>null</GetTemplateSummaryResponse>">>)
+    ))},
+
+    ExpectedParams = [{"StackName", something}],
```

Comment:
why atom `something` is used here instead of string as it is supposed to be? this is confusing, although it works since all the mecks :) 

please change it to be string anyway